src/rules/preferReadonlyRule.ts

```diff
@@ -0,0 +1,199 @@
+        if (this.scope === undefined) {
+            ts.forEachChild(node, this.visitNode);
+            return;
+        }
+
+        this.scope.enterNonConstructorScope();
+        ts.forEachChild(node, this.visitNode);
+        this.scope.exitNonConstructorScope();
+    }
+
+    private handleClassDeclarationOrExpression(node: ts.ClassLikeDeclaration) {
+        const parentScope = this.scope;
+        const childScope = this.scope = new ClassScope(node, this.typeChecker);
+
+        ts.forEachChild(node, this.visitNode);
+
+        this.finalizeScope(childScope);
+        this.scope = parentScope;
+    }
+
+    private handleConstructor(node: ts.ConstructorDeclaration) {
+        if (this.scope !== undefined) {
+            this.scope.enterConstructor();
+
+            for (const parameter of node.parameters) {
+                if (utils.isModifierFlagSet(parameter, ts.ModifierFlags.Private)) {
+                    this.scope.addDeclaredVariable(parameter);
+                }
+            }
+        }
+
+        ts.forEachChild(node, this.visitNode);
+
+        if (this.scope !== undefined) {
+            this.scope.exitConstructor();
+        }
+    }
+
+    private handlePropertyDeclaration(node: ts.PropertyDeclaration) {
+        if (this.scope !== undefined) {
+            this.scope.addDeclaredVariable(node);
+        }
+
+        ts.forEachChild(node, this.visitNode);
+    }
+
+    private handlePropertyAccessExpression(node: ts.PropertyAccessExpression, parent: ts.Node) {
+        switch (parent.kind) {
+            case ts.SyntaxKind.BinaryExpression:
+                this.handleParentBinaryExpression(node, parent as ts.BinaryExpression);
+                break;
+
+            case ts.SyntaxKind.DeleteExpression:
+                this.handleDeleteExpression(node);
+                break;
+
+            case ts.SyntaxKind.PostfixUnaryExpression:
+            case ts.SyntaxKind.PrefixUnaryExpression:
+                this.handleParentPostfixOrPrefixUnaryExpression(parent as ts.PostfixUnaryExpression | ts.PrefixUnaryExpression);
+        }
+
+        ts.forEachChild(node, this.visitNode);
+    }
+
+    private handleParentBinaryExpression(node: ts.PropertyAccessExpression, parent: ts.BinaryExpression) {
+        if (this.scope !== undefined && parent.left === node && utils.isAssignmentKind(parent.operatorToken.kind)) {
+            this.scope.addVariableModification(node);
+        }
+    }
+
+    private handleParentPostfixOrPrefixUnaryExpression(node: ts.PostfixUnaryExpression | ts.PrefixUnaryExpression) {
+        if (this.scope === undefined
+            || (node.operator !== ts.SyntaxKind.PlusPlusToken && node.operator !== ts.SyntaxKind.MinusMinusToken)) {
+            return;
+        }
+
+        this.scope.addVariableModification(node.operand as ts.PropertyAccessExpression);
+    }
+
+    private handleDeleteExpression(node: ts.PropertyAccessExpression) {
+        if (this.scope !== undefined) {
```

Comment:
you could move this check to `handlePropertyAccessExpression`, so you don't have to check again here and in the other methods